### PR TITLE
feat: add C ndarray interface and refactor implementation for `stats/base/sminabs`

### PR DESCRIPTION
**Pull request by aayush0325**
Resolves None

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C ndarray interface for `stats/base/sminabs`
-   refactors JavaScript implementation
-   updates tests, docs, examples and benchmarks.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves None

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/stats/base/sminabs/README.md [hunk 1]
```diff
@@ -170,7 +272,7 @@ console.log( v );
 
 -   <span class="package-name">[`@stdlib/stats/base/dminabs`][@stdlib/stats/base/dminabs]</span><span class="delimiter">: </span><span class="description">calculate the minimum absolute value of a double-precision floating-point strided array.</span>
 -   <span class="package-name">[`@stdlib/stats/base/minabs`][@stdlib/stats/base/minabs]</span><span class="delimiter">: </span><span class="description">calculate the minimum absolute value of a strided array.</span>
--   <span class="package-name">[`@stdlib/stats/base/smaxabs`][@stdlib/stats/base/smaxabs]</span><span class="delimiter">: </span><span class="description">calculate the maximum absolute value of a single-precision floating-point strided array.</span>
+-   <span class="package-name">[`@stdlib/stats/base/sminabsabs`][@stdlib/stats/base/sminabsabs]</span><span class="delimiter">: </span><span class="description">calculate the maximum absolute value of a single-precision floating-point strided array.</span>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/sminabs/README.md]:
```suggestion
-   <span class="package-name">[`@stdlib/stats/base/smaxabs`][@stdlib/stats/base/smaxabs]</span><span class="delimiter">: </span><span class="description">calculate the maximum absolute value of a single-precision floating-point strided array.</span>
```

File: lib/node_modules/@stdlib/stats/base/sminabs/README.md [hunk 2]
```diff
@@ -192,7 +294,7 @@ console.log( v );
 
 [@stdlib/stats/base/minabs]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/base/minabs
 
-[@stdlib/stats/base/smaxabs]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/base/smaxabs
+[@stdlib/stats/base/sminabsabs]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/base/sminabsabs
```
Comment (kgryte) [inline lib/node_modules/@stdlib/stats/base/sminabs/README.md]:
```suggestion
[@stdlib/stats/base/smaxabs]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/base/smaxabs
```

Comment (aayush0325) [reply lib/node_modules/@stdlib/stats/base/sminabs/README.md]: ah sorry about that!

**Review by kgryte — APPROVED** [review]
LGTM.

**Pull request merged**